cellprofiler/modules/measureobjectsizeshape.py

```diff
@@ -336,130 +476,138 @@ def run(self, workspace):
-            msolidity = numpy.zeros(nobjects)
-            euler = numpy.zeros(nobjects)
+            desired_properties = [
+                "label",
+                "image",
+                "area",
+                "perimeter",
+                "bbox",
+                "bbox_area",
+                "major_axis_length",
+                "minor_axis_length",
+                "orientation",
+                "centroid",
+                "equivalent_diameter",
+                "extent",
+                "eccentricity",
+                "solidity",
+                "euler_number",
+            ]
+            if self.calculate_advanced.value:
+                desired_properties += [
+                    "inertia_tensor",
+                    "inertia_tensor_eigvals",
+                    "moments",
+                    "moments_central",
+                    "moments_hu",
+                    "moments_normalized",
+                ]
+        else:
+            desired_properties = [
+                "label",
+                "image",
+                "area",
+                "centroid",
+                "bbox",
+                "bbox_area",
+                "major_axis_length",
+                "minor_axis_length",
+                "extent",
+                "equivalent_diameter",
+                "euler_number",
+            ]
+            if self.calculate_advanced.value:
+                desired_properties += [
+                    "solidity",
+                ]
+
+        # Check for overlapping object sets
+        if not objects.overlapping():
+            features_to_record = self.analyse_objects(objects, desired_properties)
+        else:
+            # Objects are overlapping, process as single arrays
+            coords_array = objects.ijv
+            features_to_record = {}
+            for label in objects.indices:
+                omap = numpy.zeros(objects.shape)
+                ocoords = coords_array[coords_array[:, 2] == label, 0:2]
+                numpy.put(omap, numpy.ravel_multi_index(ocoords.T, omap.shape), 1)
+                tempobject = cellprofiler_core.object.Objects()
+                tempobject.segmented = omap
+                buffer = self.analyse_objects(tempobject, desired_properties)
+                for f, m in buffer.items():
+                    if f in features_to_record:
+                        features_to_record[f] = numpy.concatenate((features_to_record[f], m))
+                    else:
+                        features_to_record[f] = m
+        for f, m in features_to_record.items():
+            self.record_measurement(workspace, object_name, f, m)
+
+    def analyse_objects(self, objects, desired_properties):
+        """Computing the measurements for a single map of objects"""
+        labels = objects.segmented
+        nobjects = len(objects.indices)
+        if len(objects.shape) == 2:
+            props = skimage.measure.regionprops_table(labels, properties=desired_properties)
```

Comment:
Why `regionprops_table` rather than `region props`?

Reply:
When given the `properties` parameter the table method will only calculate the measurements we're actually going to use. `_table` also saves us having to construct output arrays and iterate over the objects to extract their properties.

As a point of interest, Skimage 0.18 is going to add an `extra_properties` parameter to regionprops which will allow custom properties to be calculated. That might be a good pathway to replace the rest of the Centrosome functionality.